drivers/serial/uart_mcux_flexcomm.c

```diff
@@ -342,6 +379,636 @@ static int mcux_flexcomm_uart_config_get(const struct device *dev,
+	/* Now that this transfer was finished, reset tracking variables */
+	data->rx_data.count = 0;
+	data->rx_data.offset = 0;
+}
+
+#if defined(CONFIG_SOC_SERIES_IMX_RT5XX) || defined(CONFIG_SOC_SERIES_IMX_RT6XX)
+static uint32_t fc_uart_calc_inmux_connection(uint8_t channel, DMA_Type *base)
+{
+	uint32_t chmux_avl = 0;
+	uint32_t chmux_sel = 0;
+	uint32_t chmux_val = 0;
+
+#if defined(CONFIG_SOC_SERIES_IMX_RT5XX)
+	uint32_t chmux_sel_id = 0;
+
+	if (base == (DMA_Type *)DMA0_BASE) {
+		chmux_sel_id = DMA0_CHMUX_SEL0_ID;
+	} else if (base == (DMA_Type *)DMA1_BASE) {
+		chmux_sel_id = DMA1_CHMUX_SEL0_ID;
+	}
+
+
+	if (channel >= 16 && !(channel >= 24 && channel <= 27)) {
+		chmux_avl = 1 << CHMUX_AVL_SHIFT;
+	} else {
+		chmux_avl = 0;
+	}
+
+	/* 1 for flexcomm */
+	chmux_val = 1 << CHMUX_VAL_SHIFT;
+
+
+	if (channel <= 15 || (channel >= 24 && channel <= 27)) {
+		chmux_sel = 0;
+	} else if (channel >= 16 && channel <= 23) {
+		chmux_sel = (chmux_sel_id + 4 * (channel - 16))
+				<< CHMUX_OFF_SHIFT;
+	} else {
+		chmux_sel = (chmux_sel_id + 4 * (channel - 20))
+				<< CHMUX_OFF_SHIFT;
+	}
+
+#endif /* RT5xx */
+
+	uint32_t req_en_id = 0;
+
+	if (base == (DMA_Type *)DMA0_BASE) {
+		req_en_id = DMA0_REQ_ENA0_ID;
+	} else if (base == (DMA_Type *)DMA1_BASE) {
+		req_en_id = DMA1_REQ_ENA0_ID;
+	}
+
+
+	uint32_t en_val;
+
+	if (channel <= 31) {
+		en_val = channel + (req_en_id << ENA_SHIFT);
+	} else {
+		en_val = (channel - 32) + ((req_en_id + 4) << ENA_SHIFT);
+	}
+
+
+	uint32_t ret = en_val + chmux_avl + chmux_val + chmux_sel;
+
+	return ret;
+}
+#endif /* RT 3-digit */
+
+
+static int flexcomm_uart_async_init(const struct device *dev)
+{
+	const struct mcux_flexcomm_config *config = dev->config;
+	struct mcux_flexcomm_data *data = dev->data;
+
+	if (config->rx_dma.dev == NULL ||
+		config->tx_dma.dev == NULL) {
+		return -ENODEV;
+	}
+
+	if (!device_is_ready(config->rx_dma.dev) ||
+		!device_is_ready(config->tx_dma.dev)) {
+		return -ENODEV;
+	}
+
+	/* Disable DMA requests */
+	USART_EnableTxDMA(config->base, false);
+	USART_EnableRxDMA(config->base, false);
+
+	/* Route DMA requests */
+#if defined(CONFIG_SOC_SERIES_IMX_RT5XX) || defined(CONFIG_SOC_SERIES_IMX_RT6XX)
+	/* RT 3 digit uses input mux to route DMA requests from
+	 * the UART peripheral to a hardware designated DMA channel
+	 */
+	INPUTMUX_Init(INPUTMUX);
+	INPUTMUX_EnableSignal(INPUTMUX,
+		fc_uart_calc_inmux_connection(config->rx_dma.channel,
+					config->rx_dma.base), true);
+	INPUTMUX_EnableSignal(INPUTMUX,
+		fc_uart_calc_inmux_connection(config->tx_dma.channel,
+					config->tx_dma.base), true);
+	INPUTMUX_Deinit(INPUTMUX);
+#endif /* RT5xx and RT6xx */
+
+	/* Init work objects for RX and TX timeouts */
+	k_work_init_delayable(&data->tx_data.timeout_work,
+			config->tx_timeout_func);
+	k_work_init_delayable(&data->rx_data.timeout_work,
+			config->rx_timeout_func);
+
+	/* TX FIFO interrupt; configure to interrupt when empty */
+	config->base->FIFOTRIG &= ~USART_FIFOSTAT_TXLVL_MASK;
```

Comment:
This is already set in the USART_Init call, by the `txWatermark` field in the configuration structure (which defaults to 0)

Reply:
I know, I specifically only included this line for clarity. Do you think it should be removed

Reply:
Ideally yes, remove it and add a comment that clearing the FIFOTRIG register is not required.